### ISSUE TITLE
Realtek RTL8821CE kernel panics the system (virtio)

### ISSUE THREAD

**Issue by Chicchi7393** [home-assistant/operating-system#3561]
### Describe the issue you are experiencing

Hi,
I was interested to do a pci passthrough of my wifi+bt card, the rtl8821ce, to my libvirt HAOS, but on boot it kernel panics spitting out this error on the serial console

```
[    1.996135] BUG: kernel NULL pointer dereference, address: 000000000000003c
[    1.996953] #PF: supervisor read access in kernel mode
[    1.997572] #PF: error_code(0x0000) - not-present page
[    1.998179] PGD 0 P4D 0 
[    1.998471] Oops: 0000 [#1] PREEMPT SMP NOPTI
[    1.998958] CPU: 1 PID: 169 Comm: (udev-worker) Not tainted 6.6.46-haos #1
[    1.999805] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 0.0.0 02/06/2015
[    2.000764] RIP: 0010:rtw_pci_probe+0x91c/0x1b30 [rtw88_pci]
[    2.001312] Code: 48 c7 c6 48 13 32 c0 e8 42 92 d3 cc 49 8b 7c 24 08 48 c7 c6 98 13 32 c0 e8 31 92 d3 cc 48 89 ef e8 d9 9c 9c cc e9 3d fd ff ff <66> 41 81 7e 3c 86 80 0f 85 bd fd ff ff 41 c6 84 24 18 7b 00 00 01
[    2.003318] RSP: 0018:ffffc900006d7b28 EFLAGS: 00010246
[    2.003966] RAX: 0000000000000000 RBX: ffffffffc0297040 RCX: 0000000000000000
[    2.004887] RDX: ffff88810b51548b RSI: ffff88810b512b62 RDI: ffff88810b512376
[    2.005814] RBP: ffff888100a0b000 R08: ffff88810b512376 R09: 0000000000000003
[    2.006736] R10: ffff88810b51548b R11: ffff88810b512b62 R12: ffff88810b512040
[    2.007639] R13: ffff88810b517778 R14: 0000000000000000 R15: 0000000000000200
[    2.008552] FS:  00007f8ad1fa9840(0000) GS:ffff8881b9c80000(0000) knlGS:0000000000000000
[    2.009612] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    2.010377] CR2: 000000000000003c CR3: 0000000107c60005 CR4: 0000000000370ee0
[    2.011274] Call Trace:
[    2.011614]  <TASK>
[    2.011905]  ? __die+0x1e/0x60
[    2.012308]  ? page_fault_oops+0x178/0x490
[    2.012836]  ? exc_page_fault+0x6d/0x150
[    2.013337]  ? asm_exc_page_fault+0x26/0x30
[    2.013903]  ? rtw_pci_probe+0x91c/0x1b30 [rtw88_pci]
[    2.014574]  ? rtw_pci_probe+0x6cd/0x1b30 [rtw88_pci]
[    2.015226]  local_pci_probe+0x3a/0x90
[    2.015765]  pci_device_probe+0xab/0x180
[    2.016302]  really_probe+0x15a/0x2c0
[    2.016805]  ? __pfx___driver_attach+0x10/0x10
[    2.017377]  __driver_probe_device+0x6e/0x110
[    2.018000]  driver_probe_device+0x1a/0xe0
[    2.018563]  __driver_attach+0x83/0x180
[    2.019109]  bus_for_each_dev+0x80/0xd0
[    2.019631]  bus_add_driver+0xe3/0x210
[    2.020147]  driver_register+0x54/0x100
[    2.020665]  ? __pfx_rtw_8821ce_driver_init+0x10/0x10 [rtw88_8821ce]
[    2.021544]  do_one_initcall+0x55/0x230
[    2.022136]  do_init_module+0x5b/0x200
[    2.022653]  init_module_from_file+0x81/0xc0
[    2.023294]  idempotent_init_module+0x11b/0x310
[    2.023906]  __x64_sys_finit_module+0x51/0x90
[    2.024524]  do_syscall_64+0x38/0x80
[    2.025038]  entry_SYSCALL_64_after_hwframe+0x78/0xe2
[    2.025707] RIP: 0033:0x7f8ad27be6a9
[    2.026142] Code: ff c3 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 44 00 00 48 89 f8 48 89 f7 48 89 d6 48 89 ca 4d 89 c2 4d 89 c8 4c 8b 4c 24 08 0f 05 <48> 3d 01 f0 ff ff 73 01 c3 48 8b 0d 47 97 0c 00 f7 d8 64 89 01 48
[    2.028585] RSP: 002b:00007ffc14e1d758 EFLAGS: 00000246 ORIG_RAX: 0000000000000139
[    2.029523] RAX: ffffffffffffffda RBX: 0000000000000000 RCX: 00007f8ad27be6a9
[    2.030374] RDX: 0000000000000000 RSI: 00007f8ad28aaafd RDI: 0000000000000013
[    2.031182] RBP: 000055d40f9c5bd0 R08: 0000000000000040 R09: 0000000000000000
[    2.032053] R10: 0000000000000038 R11: 0000000000000246 R12: 00007f8ad28aaafd
[    2.033009] R13: 0000000000020000 R14: 000055d40f961bc0 R15: 0000000000000000
[    2.033939]  </TASK>
[    2.034215] Modules linked in: sch_fq_codel rtw88_8821ce(+) rtw88_8821c rtw88_pci rtw88_core mac80211 libarc4 cfg80211 cdc_acm virtio_balloon dmi_sysfs qemu_fw_cfg
[    2.036212] CR2: 000000000000003c
[    2.036667] ---[ end trace 0000000000000000 ]---
[    2.037276] RIP: 0010:rtw_pci_probe+0x91c/0x1b30 [rtw88_pci]
[    2.038001] Code: 48 c7 c6 48 13 32 c0 e8 42 92 d3 cc 49 8b 7c 24 08 48 c7 c6 98 13 32 c0 e8 31 92 d3 cc 48 89 ef e8 d9 9c 9c cc e9 3d fd ff ff <66> 41 81 7e 3c 86 80 0f 85 bd fd ff ff 41 c6 84 24 18 7b 00 00 01
[    2.040217] RSP: 0018:ffffc900006d7b28 EFLAGS: 00010246
[    2.040972] RAX: 0000000000000000 RBX: ffffffffc0297040 RCX: 0000000000000000
[    2.042034] RDX: ffff88810b51548b RSI: ffff88810b512b62 RDI: ffff88810b512376
[    2.043053] RBP: ffff888100a0b000 R08: ffff88810b512376 R09: 0000000000000003
[    2.044077] R10: ffff88810b51548b R11: ffff88810b512b62 R12: ffff88810b512040
[    2.045096] R13: ffff88810b517778 R14: 0000000000000000 R15: 0000000000000200
[    2.046144] FS:  00007f8ad1fa9840(0000) GS:ffff8881b9c80000(0000) knlGS:0000000000000000
[    2.047321] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    2.048167] CR2: 000000000000003c CR3: 0000000107c60005 CR4: 0000000000370ee0
[    2.049267] note: (udev-worker)[169] exited with irqs disabled
[    2.168197] BUG: unable to handle page fault for address: ffffc900006d7e98
[    2.169057] #PF: supervisor read access in kernel mode
[    2.169709] #PF: error_code(0x0000) - not-present page
[    2.170357] PGD 100000067 P4D 100000067 PUD 10010f067 PMD 101f93067 PTE 0
[    2.171265] Oops: 0000 [#2] PREEMPT SMP NOPTI
[    2.171878] CPU: 1 PID: 316 Comm: (udev-worker) Tainted: G      D            6.6.46-haos #1
[    2.173027] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 0.0.0 02/06/2015
[    2.174147] RIP: 0010:idempotent_init_module+0xc2/0x310
[    2.174858] Code: 32 fc 00 49 c1 ed 38 48 89 d8 4a 8b 0c ed 60 b0 df 8e 4e 8d 24 ed 60 b0 df 8e 48 8d 51 f8 48 85 c9 48 0f 45 c2 48 85 c0 74 1b <4c> 3b 38 0f 84 e1 00 00 00 48 8b 40 08 48 85 c0 74 09 48 83 e8 08
[    2.177169] RSP: 0018:ffffc90000797e90 EFLAGS: 00010282
[    2.177951] RAX: ffffc900006d7e98 RBX: 0000000000000000 RCX: ffffc900006d7ea0
[    2.178924] RDX: ffffc900006d7e98 RSI: ffffffff8df50e7b RDI: ffffffff8edfb044
[    2.179674] RBP: 0000000000000000 R08: ffffffff8e25b8e0 R09: 0000000000000000
[    2.180317] R10: 0000000000000000 R11: 0000000000000001 R12: ffffffff8edfb1d8
[    2.181017] R13: 000000000000002f R14: ffff88810b43c300 R15: ffff8881055eb6a8
[    2.181758] FS:  00007f8ad1fa9840(0000) GS:ffff8881b9c80000(0000) knlGS:0000000000000000
[    2.182634] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    2.183236] CR2: ffffc900006d7e98 CR3: 0000000108bcc003 CR4: 0000000000370ee0
[    2.184263] Call Trace:
[    2.184561]  <TASK>
[    2.184828]  ? __die+0x1e/0x60
[    2.185246]  ? page_fault_oops+0x178/0x490
[    2.185813]  ? search_extable+0x26/0x30
[    2.186221]  ? idempotent_init_module+0xc2/0x310
[    2.186867]  ? search_module_extables+0x14/0x50
[    2.187558]  ? exc_page_fault+0xb5/0x150
[    2.188101]  ? asm_exc_page_fault+0x26/0x30
[    2.188717]  ? idempotent_init_module+0xc2/0x310
[    2.189347]  __x64_sys_finit_module+0x51/0x90
[    2.189985]  do_syscall_64+0x38/0x80
[    2.190564]  entry_SYSCALL_64_after_hwframe+0x78/0xe2
[    2.191177] RIP: 0033:0x7f8ad27be6a9
[    2.191685] Code: ff c3 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 44 00 00 48 89 f8 48 89 f7 48 89 d6 48 89 ca 4d 89 c2 4d 89 c8 4c 8b 4c 24 08 0f 05 <48> 3d 01 f0 ff ff 73 01 c3 48 8b 0d 47 97 0c 00 f7 d8 64 89 01 48
[    2.194041] RSP: 002b:00007ffc14e1d758 EFLAGS: 00000246 ORIG_RAX: 0000000000000139
[    2.194997] RAX: ffffffffffffffda RBX: 0000000000000000 RCX: 00007f8ad27be6a9
[    2.195919] RDX: 0000000000000000 RSI: 00007f8ad28aaafd RDI: 0000000000000006
[    2.196798] RBP: 000055d40f98ff70 R08: 0000000000000060 R09: 00007f8ad2bb76c0
[    2.197557] R10: 0000000000000038 R11: 0000000000000246 R12: 00007f8ad28aaafd
[    2.198245] R13: 0000000000020000 R14: 000055d40f961c60 R15: 0000000000000000
[    2.199303]  </TASK>
[    2.199639] Modules linked in: sch_fq_codel rtw88_8821ce(+) rtw88_8821c rtw88_pci rtw88_core mac80211 libarc4 cfg80211 cdc_acm virtio_balloon dmi_sysfs qemu_fw_cfg
[    2.201751] CR2: ffffc900006d7e98
[    2.202338] ---[ end trace 0000000000000000 ]---
[    2.203024] RIP: 0010:rtw_pci_probe+0x91c/0x1b30 [rtw88_pci]
[    2.203914] Code: 48 c7 c6 48 13 32 c0 e8 42 92 d3 cc 49 8b 7c 24 08 48 c7 c6 98 13 32 c0 e8 31 92 d3 cc 48 89 ef e8 d9 9c 9c cc e9 3d fd ff ff <66> 41 81 7e 3c 86 80 0f 85 bd fd ff ff 41 c6 84 24 18 7b 00 00 01
[    2.206883] RSP: 0018:ffffc900006d7b28 EFLAGS: 00010246
[    2.207715] RAX: 0000000000000000 RBX: ffffffffc0297040 RCX: 0000000000000000
[    2.208830] RDX: ffff88810b51548b RSI: ffff88810b512b62 RDI: ffff88810b512376
[    2.209995] RBP: ffff888100a0b000 R08: ffff88810b512376 R09: 0000000000000003
[    2.211155] R10: ffff88810b51548b R11: ffff88810b512b62 R12: ffff88810b512040
[    2.212437] R13: ffff88810b517778 R14: 0000000000000000 R15: 0000000000000200
[    2.213557] FS:  00007f8ad1fa9840(0000) GS:ffff8881b9c80000(0000) knlGS:0000000000000000
[    2.215035] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[    2.216000] CR2: ffffc900006d7e98 CR3: 0000000108bcc003 CR4: 0000000000370ee0
[    2.217169] note: (udev-worker)[316] exited with irqs disabled
[    2.218230] note: (udev-worker)[316] exited with preempt_count 1
[    2.524447] EXT4-fs (sda8): mounted filesystem 9b0bfdd2-eb14-4b28-97a0-c63274ea5322 r/w with ordered data mode. Quota mode: none.
[    2.581546] EXT4-fs (sda8): resizing filesystem from 8210171 to 8210171 blocks
[    2.586345] EXT4-fs (sda8): Invalid checksum for backup superblock 32768
[    2.590543] EXT4-fs (sda8): Invalid checksum for backup superblock 98304
[    2.594663] EXT4-fs (sda8): Invalid checksum for backup superblock 163840
[    2.599022] EXT4-fs (sda8): Invalid checksum for backup superblock 229376
[    2.602939] EXT4-fs (sda8): Invalid checksum for backup superblock 294912
[    2.607753] EXT4-fs (sda8): Invalid checksum for backup superblock 819200
[    2.612759] EXT4-fs (sda8): Invalid checksum for backup superblock 884736
[    2.617925] EXT4-fs (sda8): Invalid checksum for backup superblock 1605632
[    2.693820] Adding 2002568k swap on /mnt/data/swapfile.  Priority:-2 extents:45 across:17665672k 
[    2.722136] EXT4-fs (zram2): mounted filesystem 2042bf62-1f47-4488-b67a-f7fdb93ae431 r/w without journal. Quota mode: none.
[    2.846237] systemd-journald[126]: Received client request to flush runtime journal.
[    2.851043] systemd-journald[126]: File /var/log/journal/1bb33cbda5944423b1c1ffae2177bd04/system.journal corrupted or uncleanly shut down, renaming and replacing.
[    2.984513] audit: type=1334 audit(1724596285.565:4): prog-id=8 op=LOAD
[    3.084451] audit: type=1334 audit(1724596285.665:5): prog-id=9 op=LOAD
[    3.085302] audit: type=1334 audit(1724596285.666:6): prog-id=10 op=LOAD
[    3.086746] audit: type=1334 audit(1724596285.666:7): prog-id=11 op=LOAD
[    3.115618] audit: type=1334 audit(1724596285.696:8): prog-id=12 op=LOAD
[    3.116535] audit: type=1334 audit(1724596285.696:9): prog-id=13 op=LOAD
[    3.117457] audit: type=1334 audit(1724596285.696:10): prog-id=14 op=LOAD
[   24.564488] rcu: INFO: rcu_preempt self-detected stall on CPU
[   24.565190] rcu:     0-....: (20999 ticks this GP) idle=0d94/1/0x4000000000000000 softirq=5164/5164 fqs=5076
[   24.566580] rcu:     (t=21000 jiffies g=1433 q=2056 ncpus=4)
[   24.567399] CPU: 0 PID: 533 Comm: modprobe Tainted: G      D            6.6.46-haos #1
[   24.568398] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 0.0.0 02/06/2015
[   24.569486] RIP: 0010:pv_native_safe_halt+0xf/0x20
[   24.570208] Code: 0b 66 2e 0f 1f 84 00 00 00 00 00 90 90 90 90 90 90 90 90 90 90 90 90 90 90 90 90 f3 0f 1e fa 66 90 0f 00 2d 15 66 31 00 fb f4 <c3> cc cc cc cc 66 2e 0f 1f 84 00 00 00 00 00 66 90 90 90 90 90 90
[   24.572503] RSP: 0018:ffffc9000170be38 EFLAGS: 00000246
[   24.573181] RAX: 0000000000000003 RBX: 0000000000040000 RCX: 0000000000000008
[   24.574088] RDX: 0000000000000000 RSI: 0000000000000003 RDI: ffffffff8edfb044
[   24.575006] RBP: 0000000000000001 R08: 0000000000000064 R09: 0000000000000000
[   24.575903] R10: ffff8881b9c2f780 R11: 0000000000000001 R12: 0000000000000100
[   24.576797] R13: ffffffff8edfb044 R14: ffff8881b9c2f780 R15: 0000000000000000
[   24.577686] FS:  00007f2e2f82a740(0000) GS:ffff8881b9c00000(0000) knlGS:0000000000000000
[   24.578755] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[   24.579500] CR2: 00007f2e2f99f445 CR3: 000000011b2d4005 CR4: 0000000000370ef0
[   24.580408] Call Trace:
[   24.580730]  <IRQ>
[   24.580991]  ? rcu_dump_cpu_stacks+0xe5/0x170
[   24.581544]  ? rcu_sched_clock_irq+0x64f/0x10e0
[   24.582136]  ? timekeeping_update+0xaa/0x280
[   24.582633]  ? _raw_spin_unlock_irqrestore+0x9/0x30
[   24.583269]  ? timekeeping_advance+0x372/0x590
[   24.583832]  ? update_process_times+0x6f/0xb0
[   24.584435]  ? tick_sched_handle+0x2e/0x40
[   24.584957]  ? tick_sched_timer+0x6a/0x80
[   24.585471]  ? __pfx_tick_sched_timer+0x10/0x10
[   24.586041]  ? __hrtimer_run_queues+0x10a/0x2a0
[   24.586632]  ? hrtimer_interrupt+0xf3/0x230
[   24.587177]  ? __sysvec_apic_timer_interrupt+0x48/0x140
[   24.587833]  ? sysvec_apic_timer_interrupt+0x65/0x80
[   24.588522]  </IRQ>
[   24.588804]  <TASK>
[   24.589079]  ? asm_sysvec_apic_timer_interrupt+0x1a/0x20
[   24.589736]  ? pv_native_safe_halt+0xf/0x20
[   24.590279]  kvm_wait+0x4e/0x60
[   24.590686]  __pv_queued_spin_lock_slowpath+0x32e/0x370
[   24.591379]  _raw_spin_lock+0x24/0x30
[   24.591851]  idempotent_init_module+0x9b/0x310
[   24.592457]  __x64_sys_finit_module+0x51/0x90
[   24.593028]  do_syscall_64+0x38/0x80
[   24.593526]  entry_SYSCALL_64_after_hwframe+0x78/0xe2
[   24.594163] RIP: 0033:0x7f2e2f92f6a9
[   24.594672] Code: ff c3 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 44 00 00 48 89 f8 48 89 f7 48 89 d6 48 89 ca 4d 89 c2 4d 89 c8 4c 8b 4c 24 08 0f 05 <48> 3d 01 f0 ff ff 73 01 c3 48 8b 0d 47 97 0c 00 f7 d8 64 89 01 48
[   24.597042] RSP: 002b:00007fff37333278 EFLAGS: 00000246 ORIG_RAX: 0000000000000139
[   24.598014] RAX: ffffffffffffffda RBX: 0000000000000000 RCX: 00007f2e2f92f6a9
[   24.598947] RDX: 0000000000000000 RSI: 00005630094984be RDI: 0000000000000000
[   24.599879] RBP: 0000563009958e80 R08: 0000000000000060 R09: 0000000000000000
[   24.600811] R10: 0000000000000038 R11: 0000000000000246 R12: 00005630094984be
[   24.601776] R13: 0000000000040000 R14: 0000563009958dc0 R15: 0000000000000000
[   24.602721]  </TASK>
[   54.142653] rcu: INFO: rcu_preempt detected expedited stalls on CPUs/tasks: { 0-.... } 21203 jiffies s: 321 root: 0x1/.
[   54.143741] rcu: blocking rcu_node structures (internal RCU debug):
[   54.144381] Sending NMI from CPU 2 to CPUs 0:
[   54.144931] NMI backtrace for cpu 0
[   54.144940] CPU: 0 PID: 533 Comm: modprobe Tainted: G      D            6.6.46-haos #1
[   54.144943] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 0.0.0 02/06/2015
[   54.144947] RIP: 0010:__pv_queued_spin_lock_slowpath+0xf7/0x370
[   54.144956] Code: 31 c0 bd 01 00 00 00 41 bc 00 01 00 00 3c 02 41 0f 94 c0 4c 89 04 24 41 c6 46 14 00 ba 00 80 00 00 41 c6 45 01 01 eb 0b f3 90 <83> ea 01 0f 84 10 02 00 00 41 0f b6 45 00 84 c0 75 ec 44 89 e0 f0
[   54.144961] RSP: 0018:ffffc9000170be48 EFLAGS: 00000206
[   54.144964] RAX: 0000000000000003 RBX: 0000000000040000 RCX: 0000000000000008
[   54.144965] RDX: 00000000000051cf RSI: 0000000000000003 RDI: ffffffff8edfb044
[   54.144967] RBP: 0000000000000001 R08: 0000000000000064 R09: 0000000000000000
[   54.144968] R10: ffff8881b9c2f780 R11: 0000000000000001 R12: 0000000000000100
[   54.144969] R13: ffffffff8edfb044 R14: ffff8881b9c2f780 R15: 0000000000000000
[   54.144971] FS:  00007f2e2f82a740(0000) GS:ffff8881b9c00000(0000) knlGS:0000000000000000
[   54.144972] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[   54.144974] CR2: 00007f2e2f99f445 CR3: 000000011b2d4005 CR4: 0000000000370ef0
[   54.144977] Call Trace:
[   54.144980]  <NMI>
[   54.144982]  ? nmi_cpu_backtrace+0x99/0x110
[   54.144987]  ? nmi_cpu_backtrace_handler+0xc/0x20
[   54.144990]  ? nmi_handle+0x59/0x150
[   54.144993]  ? default_do_nmi+0x40/0x100
[   54.144996]  ? exc_nmi+0x102/0x130
[   54.144997]  ? end_repeat_nmi+0x16/0x67
[   54.145001]  ? __pv_queued_spin_lock_slowpath+0xf7/0x370
[   54.145003]  ? __pv_queued_spin_lock_slowpath+0xf7/0x370
[   54.145005]  ? __pv_queued_spin_lock_slowpath+0xf7/0x370
[   54.145007]  </NMI>
[   54.145007]  <TASK>
[   54.145008]  _raw_spin_lock+0x24/0x30
[   54.145010]  idempotent_init_module+0x9b/0x310
[   54.145013]  __x64_sys_finit_module+0x51/0x90
[   54.145016]  do_syscall_64+0x38/0x80
[   54.145017]  entry_SYSCALL_64_after_hwframe+0x78/0xe2
[   54.145019] RIP: 0033:0x7f2e2f92f6a9
[   54.145021] Code: ff c3 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 44 00 00 48 89 f8 48 89 f7 48 89 d6 48 89 ca 4d 89 c2 4d 89 c8 4c 8b 4c 24 08 0f 05 <48> 3d 01 f0 ff ff 73 01 c3 48 8b 0d 47 97 0c 00 f7 d8 64 89 01 48
[   54.145022] RSP: 002b:00007fff37333278 EFLAGS: 00000246 ORIG_RAX: 0000000000000139
[   54.145024] RAX: ffffffffffffffda RBX: 0000000000000000 RCX: 00007f2e2f92f6a9
[   54.145025] RDX: 0000000000000000 RSI: 00005630094984be RDI: 0000000000000000
[   54.145025] RBP: 0000563009958e80 R08: 0000000000000060 R09: 0000000000000000
[   54.145026] R10: 0000000000000038 R11: 0000000000000246 R12: 00005630094984be
[   54.145027] R13: 0000000000040000 R14: 0000563009958dc0 R15: 0000000000000000
[   54.145028]  </TASK>
[   87.580772] rcu: INFO: rcu_preempt self-detected stall on CPU
[   87.581540] rcu:     0-....: (83972 ticks this GP) idle=0d94/1/0x4000000000000000 softirq=5164/5164 fqs=20491
[   87.582928] rcu:     (t=84038 jiffies g=1433 q=2723 ncpus=4)
[   87.583728] CPU: 0 PID: 533 Comm: modprobe Tainted: G      D            6.6.46-haos #1
[   87.584880] Hardware name: QEMU Standard PC (i440FX + PIIX, 1996), BIOS 0.0.0 02/06/2015
[   87.586047] RIP: 0010:__pv_queued_spin_lock_slowpath+0xf7/0x370
[   87.586778] Code: 31 c0 bd 01 00 00 00 41 bc 00 01 00 00 3c 02 41 0f 94 c0 4c 89 04 24 41 c6 46 14 00 ba 00 80 00 00 41 c6 45 01 01 eb 0b f3 90 <83> ea 01 0f 84 10 02 00 00 41 0f b6 45 00 84 c0 75 ec 44 89 e0 f0
[   87.589025] RSP: 0018:ffffc9000170be48 EFLAGS: 00000206
[   87.589686] RAX: 0000000000000003 RBX: 0000000000040000 RCX: 0000000000000008
[   87.590635] RDX: 0000000000000583 RSI: 0000000000000003 RDI: ffffffff8edfb044
[   87.591532] RBP: 0000000000000001 R08: 0000000000000064 R09: 0000000000000000
[   87.592358] R10: ffff8881b9c2f780 R11: 0000000000000001 R12: 0000000000000100
[   87.593103] R13: ffffffff8edfb044 R14: ffff8881b9c2f780 R15: 0000000000000000
[   87.593849] FS:  00007f2e2f82a740(0000) GS:ffff8881b9c00000(0000) knlGS:0000000000000000
[   87.594838] CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
[   87.595516] CR2: 00007f2e2f99f445 CR3: 000000011b2d4005 CR4: 0000000000370ef0
[   87.596360] Call Trace:
[   87.596663]  <IRQ>
[   87.596924]  ? rcu_dump_cpu_stacks+0xe5/0x170
[   87.597456]  ? rcu_sched_clock_irq+0x64f/0x10e0
[   87.598036]  ? update_load_avg+0x77/0x710
[   87.598524]  ? _raw_spin_unlock_irqrestore+0x9/0x30
[   87.599115]  ? timekeeping_advance+0x372/0x590
[   87.599662]  ? update_process_times+0x6f/0xb0
[   87.600217]  ? tick_sched_handle+0x2e/0x40
[   87.600709]  ? tick_sched_timer+0x6a/0x80
[   87.601260]  ? __pfx_tick_sched_timer+0x10/0x10
[   87.601860]  ? __hrtimer_run_queues+0x10a/0x2a0
[   87.602426]  ? hrtimer_interrupt+0xf3/0x230
[   87.602965]  ? __sysvec_apic_timer_interrupt+0x48/0x140
[   87.603636]  ? sysvec_apic_timer_interrupt+0x65/0x80
[   87.604291]  </IRQ>
[   87.604562]  <TASK>
[   87.604842]  ? asm_sysvec_apic_timer_interrupt+0x1a/0x20
[   87.605517]  ? __pv_queued_spin_lock_slowpath+0xf7/0x370
[   87.606212]  ? __pv_queued_spin_lock_slowpath+0x32e/0x370
[   87.606870]  _raw_spin_lock+0x24/0x30
[   87.607360]  idempotent_init_module+0x9b/0x310
[   87.607934]  __x64_sys_finit_module+0x51/0x90
[   87.608479]  do_syscall_64+0x38/0x80
[   87.608958]  entry_SYSCALL_64_after_hwframe+0x78/0xe2
[   87.609633] RIP: 0033:0x7f2e2f92f6a9
[   87.610117] Code: ff c3 66 2e 0f 1f 84 00 00 00 00 00 0f 1f 44 00 00 48 89 f8 48 89 f7 48 89 d6 48 89 ca 4d 89 c2 4d 89 c8 4c 8b 4c 24 08 0f 05 <48> 3d 01 f0 ff ff 73 01 c3 48 8b 0d 47 97 0c 00 f7 d8 64 89 01 48
[   87.612362] RSP: 002b:00007fff37333278 EFLAGS: 00000246 ORIG_RAX: 0000000000000139
[   87.613173] RAX: ffffffffffffffda RBX: 0000000000000000 RCX: 00007f2e2f92f6a9
[   87.613997] RDX: 0000000000000000 RSI: 00005630094984be RDI: 0000000000000000
[   87.614836] RBP: 0000563009958e80 R08: 0000000000000060 R09: 0000000000000000
[   87.615623] R10: 0000000000000038 R11: 0000000000000246 R12: 00005630094984be
[   87.616504] R13: 0000000000040000 R14: 0000563009958dc0 R15: 0000000000000000
[   87.617329]  </TASK>
```

### What operating system image do you use?

generic-x86-64 (Generic UEFI capable x86-64 systems)

### What version of Home Assistant Operating System is installed?

Home Assistant OS 13.1

### Did the problem occur after upgrading the Operating System?

Yes

### Hardware details

pc-i440fx-jammy (UEFI) on libvirt 8.0.0 (qemu 6.2.0) + kvm
CPU Type: host passthrough (i have an i7 8565u)
4 vCPU
6GB ram 
32gb virtio disk
bridge virtio network card

SONOFF Zigbee 3.0 USB Dongle Plus V2 passed to the vm 

### Steps to reproduce the issue

1. add thru the pci passthrough the network card
2. reboot the system
3. system panics

### Anything in the Supervisor logs that might be useful for us?

```txt
nothing
```

### Anything in the Host logs that might be useful for us?

```txt
nothing
```

## System Information

version | core-2024.8.0
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.12.4
os_name | Linux
os_version | 6.6.46-haos
arch | x86_64
timezone | Europe/Rome
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
HACS Data | ok
GitHub API Calls Remaining | 5000
Installed Version | 2.0.0
Stage | running
Available Repositories | 1459
Downloaded Repositories | 21

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | 1 gennaio 2018 alle ore 01:00
relayer_connected | false
relayer_region | null
remote_enabled | false
remote_connected | false
alexa_enabled | false
google_enabled | false
remote_server | null
certificate_status | null
instance_id | a0ae8e9a36e14b708e7859e1437e70db
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 13.1
-- | --
update_channel | stable
supervisor_version | supervisor-2024.08.0
agent_version | 1.6.0
docker_version | 26.1.4
disk_total | 30.8 GB
disk_used | 10.9 GB
healthy | true
supported | true
host_connectivity | true
supervisor_connectivity | true
ntp_synchronized | true
virtualization | kvm
board | ova
supervisor_api | ok
version_api | ok
installed_addons | File editor (5.8.0), Matter Server (6.4.1), Terminal & SSH (9.14.0), Zigbee2MQTT (1.39.1-1), Mosquitto broker (6.4.1), Silicon Labs Multiprotocol (2.4.5), Node-RED (18.0.5), Nginx Proxy Manager (1.0.1), Traccar (0.25.0), Tailscale (0.21.0), Whisper (2.1.2), Assist Microphone (1.2.0), Silicon Labs Flasher (0.2.3), Cloudflared (5.1.18), Hyperion.NG (2.0.16), openWakeWord (1.10.0), AppDaemon (0.16.6)

</details>

<details><summary>Dashboards</summary>

dashboards | 3
-- | --
resources | 10
views | 7
mode | storage

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | 16 agosto 2024 alle ore 12:06
-- | --
current_recorder_run | 25 agosto 2024 alle ore 20:49
estimated_db_size | 57.43 MiB
database_engine | sqlite
database_version | 3.45.3

</details>

<details><summary>Sonoff</summary>

version | 3.7.3 (e240aaf)
-- | --
cloud_online | 1 / 2
local_online | 1 / 1

</details>

**Comment by sairon** [comment]
PCIe passthrough is tricky, even though it ultimately manifests as a stack trace in the guest OS, it may be caused by the host OS or any hardware settings. Can you check if it works with another Linux distribution as the guest?

**Comment by Chicchi7393** [comment]
Sorry for the late reply, I've been quite busy this past days and i kinda forgot. Anyways I've managed to boot debian testing with the passtrough, so it isn't an host os error

**Comment by Chicchi7393** [comment]
@sairon sorry to mention you, you might have missed the issue

**Comment by sairon** [comment]
I did not, I just didn't have an opportunity to look into that - until now. I am able to reproduce the issue with this card too and confirm it really works on Debian-based distros (tested latest stable Debian and Ubuntu). However, it fails the same way on Arch Linux (both with 6.6 linux-lts and 6.10 linux), and when I applied Ubuntu's kernel config to HAOS build (with some minor modification that are required for successful boot), the problem persisted, so it's unclear to me yet why it works on Debian. There are still some more things to try, for example testing of other mainstream distributions, ideally those running the same minor kernel version (i.e. 6.6.y) without many patches. That way we could start "bisecting" the configuration to find what's needed to make it working.
